apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RemoveMe/ResumableUpload.cs

```diff
@@ -0,0 +1,1077 @@
+﻿/*
+Copyright 2012 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+using System.Collections;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Reflection;
+
+using Google.Apis.Http;
+using Google.Apis.Logging;
+using Google.Apis.Media;
+using Google.Apis.Requests;
+using Google.Apis.Services;
+using Google.Apis.Testing;
+using Google.Apis.Util;
+using Google.Apis.Json;
+
+namespace Google.Apis.Upload
+{
+    public sealed class ResumableUploadSessionOptions
+    {
+        public HttpClient HttpClient { get; set; }
+
+        internal ConfigurableHttpClient ConfigurableHttpClient { get { return HttpClient as ConfigurableHttpClient; } }
+
+        public Action<HttpRequestMessage> ModifySessionUriRequest { get; set; }
+
+        public ISerializer Serializer { get; set; }
+
+        public string ServiceName { get; set; }
+    }
+
+    public class TmpResumableUpload
+    {
+        protected ConfigurableHttpClient HttpClient { get; }
+
+        private ResumableUploadSessionOptions Options { get; }
+
+        private string SignedUrl { get; set; }
```

Comment:
I think I was overthinking this one a bit too much. Also, I forgot how much the `ResumableUpload<TRequest>.GetSessionUriAsync` override's implementation was tied into its class. Ripping it out was more trouble that it was worth. We can always refactor later to make the session creation mechanism more customizable. I'm about to push another commit with changes that implement what I think you were suggesting (although I renamed GetSessionUriAsync to InitiateSessionAsync since it matches up more with the [docs](https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload#start-resumable) language). I also added the comments finally so hopefully things will be a bit clearer.